aeron-driver/src/main/java/io/aeron/driver/DriverConductor.java

```diff
@@ -382,6 +391,77 @@ void onCreatePublicationImage(
         }
     }
 
+    private static String enrichImagePublicationChannelUri(
+        final ReceiveChannelEndpoint receiveChannelEndpoint,
+        final int streamId,
+        final int initialTermId,
+        final int activeTermId,
+        final int termBufferLength,
+        final int termOffset,
+        final int senderMtuLength,
+        final int sessionId,
+        final boolean isSparse,
+        final SubscriptionParams params,
+        final CongestionControl congestionControl)
+    {
+        final ChannelUriStringBuilder channelUriStringBuilder = new ChannelUriStringBuilder(
+            receiveChannelEndpoint.udpChannel().channelUri());
+
+        channelUriStringBuilder.sessionId(sessionId);
+
+        channelUriStringBuilder.streamId(streamId);
+
+        channelUriStringBuilder.termLength(termBufferLength);
+        channelUriStringBuilder.mtu(senderMtuLength);
+
+        channelUriStringBuilder.initialTermId(initialTermId);
+        channelUriStringBuilder.termId(activeTermId);
+
+        channelUriStringBuilder.termOffset(termOffset);
+
+        channelUriStringBuilder.tether(params.isTether);
+        channelUriStringBuilder.rejoin(params.isRejoin);
+        channelUriStringBuilder.reliable(params.isReliable);
+        if (INFER != params.group)
+        {
+            channelUriStringBuilder.group(FORCE_TRUE == params.group);
+        }
+        channelUriStringBuilder.sparse(isSparse);
+
+        channelUriStringBuilder.congestionControl(congestionControlName(congestionControl));
+
+        if (params.hasSessionId)
+        {
+            channelUriStringBuilder.sessionId(params.sessionId);
+        }
+
+        channelUriStringBuilder.receiverWindowLength(params.receiverWindowLength);
+
+      //  channelUriStringBuilder.group(params.group);
+
+//        boolean hasJoinPosition = false;
+//        boolean isResponse = false;
+//        InferableBoolean group = InferableBoolean.INFER;
+//
+        return channelUriStringBuilder.toString();
+    }
+
+    private static String congestionControlName(final CongestionControl congestionControl)
+    {
+        if (congestionControl instanceof StaticWindowCongestionControl)
+        {
+            return "static";
+        }
+        else if (congestionControl instanceof CubicCongestionControl)
+        {
+            return "cubic";
+        }
+        else
+        {
+            return congestionControl.getClass().getName();
+        }
+    }
```

Comment:
Let's use `congestionControl.getClass().getName()` for simplicity. And it also aligns with the flow control handling.

Reply:
Ok. Will fix.